src/test/java/uk/gov/justice/probation/courtcaseservice/controller/CourtCaseControllerIntTest.java

```diff
@@ -373,6 +340,119 @@ void shouldReturnNotFoundForNonexistentCourt() {
         }
     }
 
+    @Nested
+    class GetCaseByCaseId {
+
+        @Test
+        void givenKnownCaseId_whenGetCase_thenReturn() {
+
+            var response = given()
+                .given()
+                .auth()
+                .oauth2(getToken())
+                .when()
+                .header("Accept", "application/json")
+                .get("/case/{caseId}", "1f93aa0a-7e46-4885-a1cb-f25a4be33a00")
+                .then()
+                .statusCode(200);
+
+            response.body("$", not(hasKey("caseNo")));
+
+            validateResponse(response, DECEMBER_14_9AM.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME), "1f93aa0a-7e46-4885-a1cb-f25a4be33a00");
+        }
+
+        @Test
+        void givenUnknownCaseId_whenGetCase_thenReturn404() {
+
+            final var NOT_FOUND_CASE_ID = "1f93bbcc-7e46-4885-a1cb-f25a4be33a56";
+
+            ErrorResponse result = given()
+                .given()
+                .auth()
+                .oauth2(getToken())
+                .when()
+                .header("Accept", "application/json")
+                .get("/case/{caseId}", NOT_FOUND_CASE_ID)
+                .then()
+                .statusCode(404)
+                .extract()
+                .body()
+                .as(ErrorResponse.class);
+
+            assertThat(result.getDeveloperMessage()).contains("Case " + NOT_FOUND_CASE_ID + " not found");
+            assertThat(result.getUserMessage()).contains("Case " + NOT_FOUND_CASE_ID + " not found");
+            assertThat(result.getStatus()).isEqualTo(404);
+        }
+
+        @Test
+        void shouldReturnNotFoundForDeletedCase() {
+
+            final var DELETED_CASE_ID = "5555559";
+
+            ErrorResponse result = given()
+                .given()
+                .auth()
+                .oauth2(getToken())
+                .when()
+                .header("Accept", "application/json")
+                .get("/case/{caseId}", DELETED_CASE_ID)
+                .then()
+                .statusCode(404)
+                .extract()
+                .body()
+                .as(ErrorResponse.class);
+
+            assertThat(result.getDeveloperMessage()).contains("Case " + DELETED_CASE_ID + " not found");
+            assertThat(result.getUserMessage()).contains("Case " + DELETED_CASE_ID + " not found");
+            assertThat(result.getStatus()).isEqualTo(404);
+        }
+
+    }
+
+    private void validateResponse(ValidatableResponse validatableResponse, String startTime, String caseId) {
+        validatableResponse
```

Comment:
I quite like this, makes the test itself nice and concise whilst still doing the heavy lifting required 👍  Perhaps we should adopt this pattern more widely?